kong/db/schema/init.lua

```diff
@@ -1736,10 +1743,12 @@ function Schema:process_auto_fields(data, context, nulls, opts)
           data[sname] = nil
           local new_values = sdata.func(value)
           if new_values then
+            local deprecation = sdata.deprecation
             for k, v in pairs(new_values) do
               if type(v) == "table" then
-                data[k] = tablex.merge(data[k] or {}, v, true)
-              else
+                data[k] = deprecation and table_merge(v, data[k])
```

Comment:
Maybe let's add a comment here to explain why deprecation is tied to shorthands?

Reply:
I am fine with asks like these. It is just that simple ask like this can create considerable amounts of work. Like restarting EE tests 10 times.

Reply:
A comment was added.